### ISSUE TITLE
When using Apipie.configuration.authorize: new_doc[:docs][:resources] is an array instead of a hash when accessing docs causing errors:

### ISSUE THREAD

**Issue by NielsKSchjoedt** [Apipie/apipie-rails#654]
When I use `config.authorize` then any documentation page (other than the `config.doc_base_url`) fails.

When accessing the documentation index page, it works: `localhost:3000/service_api_doc`, but when e.g. accessing one of the links (e.g. localhost:3000/service_api_doc/1/users.en.html) to a documentation for a controller, I get the following: 

```
I, [2018-12-13T20:11:55.366951 #48414]  INFO -- : [89a49504-ca95-4cfb-b5d7-024ec4f3e677] Processing by Apipie::ApipiesController#index as HTML
I, [2018-12-13T20:11:55.367054 #48414]  INFO -- : [89a49504-ca95-4cfb-b5d7-024ec4f3e677]   Parameters: {"version"=>"1", "resource"=>"users.html"}
I, [2018-12-13T20:12:00.758601 #48414]  INFO -- : [89a49504-ca95-4cfb-b5d7-024ec4f3e677] Completed 500 Internal Server Error in 5391ms (ActiveRecord: 11.6ms)
F, [2018-12-13T20:12:00.760367 #48414] FATAL -- : [89a49504-ca95-4cfb-b5d7-024ec4f3e677] 
NoMethodError (undefined method `[]' for nil:NilClass):
  apipie-rails (0.5.14) app/controllers/apipie/apipies_controller.rb:120:in `block in authorized_doc'
  apipie-rails (0.5.14) app/controllers/apipie/apipies_controller.rb:118:in `select'
  apipie-rails (0.5.14) app/controllers/apipie/apipies_controller.rb:118:in `authorized_doc'
  apipie-rails (0.5.14) app/controllers/apipie/apipies_controller.rb:51:in `block in index'
  actionpack (4.2.11) lib/action_controller/metal/mime_responds.rb:212:in `respond_to'
  apipie-rails (0.5.14) app/controllers/apipie/apipies_controller.rb:28:in `index'
  actionpack (4.2.11) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.2.11) lib/abstract_controller/base.rb:198:in `process_action'
  actionpack (4.2.11) lib/action_controller/metal/rendering.rb:10:in `process_action'
  actionpack (4.2.11) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.2.11) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.11) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
  activesupport (4.2.11) lib/active_support/callbacks.rb:505:in `call'
```
I tried to add a `binding.pry` to `app/controllers/apipie/apipies_controller.rb:120` it turns out, that `@doc[:docs][:resources]` is an `Array` instead of a `Hash`. The content of the array is just one hash, which seems to be what is expected in `v`. So when doing: `new_doc[:docs][:resources] = @doc[:docs][:resources].select do |k, v|` then `k` is actually what should have been in `v` and `v` is `nil` and `k` (the controller name) is missing.

As far as I can see, it seems to tie back to what happens on line:
`apipie-rails-0.5.14/lib/apipie/application.rb:298` However I have no clue why this happens?

I'm using v. 0.5.14 My configuration is:

```ruby
Apipie.configure do |config|
  config.app_name                = " Service API"
  config.app_info                = "Technical documentation for the Service API"
  config.api_base_url            = "/service_api"
  config.doc_base_url            = "/service_api_doc"
  config.api_controllers_matcher = "#{Rails.root}/app/controllers/service_api/**/*.rb"
  config.default_version         = "1"
  config.authorize = proc do |controller, _method, _doc|
    true # Even if I enable everything
  end
end
```

**Comment by NielsKSchjoedt** [comment]
I made a fix, don't know it it's the right way, but it works :-)

https://github.com/Apipie/apipie-rails/pull/655

**Comment by iNecas** [comment]
Fixed by #655
